lib/protocols.js

```diff
@@ -801,6 +801,21 @@
       }
     }
 
+    function decodeProtobufMessage(wrappedMessage){
+        var wrappedTypeId = wrappedMessage.wrappedTypeId;
+        var wrappedMessage= wrappedMessage.wrappedMessage;
+        var protostreamType = ProtostreamType.lookupProtostreamTypeById(wrappedTypeId);
+        var root = ProtobufRoot.findRootByTypeName(protostreamType.protostreamTypeName);
+        var userProtobufType= root.lookupType(protostreamType.protostreamTypeName);
```

Comment:
`lookupType` is called twice in these lines. Decoder is executed for each incoming message, better to try to optimize the code. See my comments below